### PR TITLE
[NUI] fix for a owner object to maintain the reference of owned objects

### PR DESCRIPTION
**Pull request by dongsug-song**
### Description of Change ###
[NUI] fix for a owner object to maintain the reference of owned objects
- Renderer maintains the references of Geometry and Shader
- The references are automatically released when Renderer object is disposed by DisposeQueue
- Tested by test/Tizen.NUI.Samples/Tizen.NUI.Samples/Samples/DisposeTest.cs

### API Changes ###
none

### CODE REVIEW

File: src/Tizen.NUI/src/public/Rendering/Renderer.cs [hunk 1]
```diff
@@ -587,7 +594,7 @@ public Geometry GetGeometry()
             System.IntPtr cPtr = Interop.Renderer.GetGeometry(SwigCPtr);
             Geometry ret = this.GetInstanceSafely<Geometry>(cPtr);
             if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
-            return ret;
+            return CurrentGeometry = ret;
```
Comment (myroot) [inline src/Tizen.NUI/src/public/Rendering/Renderer.cs:597]: Why don't you directly return `CurrentGeometry`?
Is it possible to update by Native layout without update of C# layer?

File: src/Tizen.NUI/src/public/Rendering/Renderer.cs [hunk 2]
```diff
@@ -27,14 +27,19 @@ namespace Tizen.NUI
     /// <since_tizen> 3 </since_tizen>
     public class Renderer : Animatable
     {
+        private Geometry CurrentGeometry = null;
```
Comment (myroot) [inline src/Tizen.NUI/src/public/Rendering/Renderer.cs:30]: You should dispose owned object when owner object was disposed. if owned object and owner are IDisposable

**Review by myroot — CHANGES_REQUESTED** [review]

File: src/Tizen.NUI/src/public/Rendering/Renderer.cs (hunk 2 above)
Comment (dongsug-song) [reply src/Tizen.NUI/src/public/Rendering/Renderer.cs:30]: I tested it is disposed automatically by DisposeQueue with DisposeTest.cs
could you check https://github.sec.samsung.net/dotnet/NUIBackend/issues/34#issuecomment-1218728 ?

Comment (dongsug-song) [reply src/Tizen.NUI/src/public/Rendering/Renderer.cs:30]: Thank you for reviewing. I added  `protected override void Dispose(bool disposing)`

File: src/Tizen.NUI/src/public/Rendering/Renderer.cs [hunk 3]
```diff
@@ -678,6 +675,25 @@ protected override void ReleaseSwigCPtr(System.Runtime.InteropServices.HandleRef
...
+            {
+                // dispose managed state (managed objects).
+                CurrentGeometry?.Dispose();
+                CurrentShader?.Dispose();
+            }
+
+            // Call the base class implementation.
+            base.Dispose(disposing);
+        }
+
```
Comment (myroot) [inline src/Tizen.NUI/src/public/Rendering/Renderer.cs:696]: 우선, https://github.sec.samsung.net/dotnet/NUIBackend/issues/34#issuecomment-1219215 
여기에서 언급했듯이, BaseHandle을 상속한 객체는 disposing이 false인 상태에서 disposing이 true로 호출되는 상황이 발생합니다. 
따라서, 차라리 IDisposable 객체에 대한 dispose는  Dispose(DisposeTypes type) 여기를 통해서 하는것이 안전할것 같습니다.
```suggestion
        protected override void Dispose(DisposeTypes type)
        {
            if (type == DisposeTypes.Explicit)
            {
                CurrentGeometry?.Dispose();
                CurrentShader?.Dispose();
            }
            base.Dispose(type);
        }
```

Comment (dongsug-song) [reply src/Tizen.NUI/src/public/Rendering/Renderer.cs]: 네. 맞습니다. 수정해 주셔서 감사합니다. 반영 하였습니다.

**Review by myroot — APPROVED** [review]

**Pull request merged**